libpoolprotocols/PoolManager.cpp

```diff
@@ -415,7 +416,12 @@ void PoolManager::rotateConnect()
     if (!m_Settings.connections.empty() && m_Settings.connections.at(m_activeConnectionIdx)->Host() != "exit")
     {
         if (p_client)
+        {
             p_client = nullptr;
+            cnote << "Retry " << m_connectionAttempt << ", sleep "<< m_Settings.delayBeforeRetry << "s before re-connect";
+            // sleep before retry 
+            std::this_thread::sleep_for(std::chrono::seconds(m_Settings.delayBeforeRetry));
```

Comment:
please do not use actual sleeps here. use a timer with async_wait  instead.

Reply:
would I need add a new timer for this? 
the pool thread is doing nothing if there is no active client, I think sleep here is reasonable.

Reply:
Yes you need a timer for it.
You probably could move it a bit further down and use a timer to do the p_client->connect in the timer callback

Based on this: https://github.com/ethereum-mining/ethminer/pull/1813#issuecomment-453002344

it will halt the whole main thread, since poolmanager has no own thread